example/bank/cmd/api/account.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/hellofresh/goengine/aggregate"
+	"github.com/hellofresh/goengine/example/bank/domain"
+	"go.uber.org/zap"
+)
+
+type contextKey string
+
+const contextKeyBankAccountID contextKey = "bankAccountID"
+
+func newAccountHandler(db *sql.DB, repo *domain.BankAccountRepository, logger *zap.Logger) *accountHandler {
+	return &accountHandler{db, repo, logger}
+}
+
+type accountHandler struct {
+	db              *sql.DB
+	bankAccountRepo *domain.BankAccountRepository
+	logger          *zap.Logger
+}
+
+func (a *accountHandler) HTTPHandler() *chi.Mux {
+	r := chi.NewRouter()
+	r.Use(middleware.NoCache)
+
+	r.Post("/", a.CreateAccount)
+
+	r.Route("/{account_id}", func(r chi.Router) {
+		r.Use(a.accountIDFromURL)
+
+		r.Get("/", a.ViewAccount)
+		r.Get("/average", a.AccountAverages)
+		r.Post("/deposit", a.AccountDeposit)
+		r.Post("/withdraw", a.AccountWithdraw)
+	})
+
+	return r
+}
+
+func (a *accountHandler) CreateAccount(w http.ResponseWriter, r *http.Request) {
+	bankAccount, err := domain.OpenBankAccount()
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		a.logger.With(zap.Error(err)).Error("failed to open a bank account")
+		return
+	}
+
+	if err := a.bankAccountRepo.Save(r.Context(), bankAccount); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		a.logger.With(zap.Error(err)).Error("failed to save a opened bank account")
+		return
+	}
+
+	w.Header().Add("Location", fmt.Sprintf("/account/%s", bankAccount.AggregateID()))
+	w.WriteHeader(http.StatusCreated)
+}
+
+func (a *accountHandler) ViewAccount(w http.ResponseWriter, r *http.Request) {
+	bankAccount := a.loadAccount(w, r)
+	if bankAccount == nil {
+		return
+	}
+	body := []byte(
+		fmt.Sprintf(`{"version":%d}`, bankAccount.AggregateVersion()),
+	)
+
+	w.Header().Add("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	if _, err := w.Write(body); err != nil {
+		a.logger.With(zap.Error(err)).Error("failed to write response")
+	}
+}
+
+func (a *accountHandler) AccountAverages(w http.ResponseWriter, r *http.Request) {
+	var (
```

Comment:
```suggestion
	var debit, credit uint
```

You can have this on one line :)